### ISSUE TITLE
Should `wgpuInstanceRequestAdapter()` imply keeping runtime alive?

### ISSUE THREAD

**Issue by hujiajie** [emscripten-core/emscripten#15246]
[`wgpuInstanceRequestAdapter()`](https://github.com/emscripten-core/emscripten/blob/e274e49bfb2022313f5697dc5560c19f5f13d81a/src/library_webgpu.js#L2048) is similar to [`emscripten_async_call()`](https://github.com/emscripten-core/emscripten/blob/e274e49bfb2022313f5697dc5560c19f5f13d81a/src/library_browser.js#L1171) in that it takes a callback, and the callback has to be scheduled after `main()` exits. The "Emscripten runtime" is automatically [kept alive](https://github.com/emscripten-core/emscripten/blob/e274e49bfb2022313f5697dc5560c19f5f13d81a/src/library.js#L3592) in `emscripten_async_call()`, so I wonder if `wgpuInstanceRequestAdapter()` (and its friends) should be responsible for this too?

Besides, I'm seeing an uncaught `'unwind'` exception when calling `emscripten_set_main_loop()` from my registered callback, while `emscripten_async_call()` has the magic to mute this exception.

**Comment by sbc100** [comment]
Yes, it sounds like those functions should use the `callUserCallback()` function internally as well as the `runtimeKeepalivePush()/runtimeKeepalivePop()` mechanism.

@kainino0x ?
